posthog-react-native/src/posthog-rn.ts

```diff
@@ -160,4 +168,65 @@ export class PostHog extends PostHogCore {
   initReactNativeNavigation(options: PostHogAutocaptureOptions): boolean {
     return withReactNativeNavigation(this, options)
   }
+
+  async captureNativeAppLifecycleEvents(): Promise<void> {
+    // See the other implementations for reference:
+    // ios: https://github.com/PostHog/posthog-ios/blob/3a6afc24d6bde730a19470d4e6b713f44d076ad9/PostHog/Classes/PHGPostHog.m#L140
+    // android: https://github.com/PostHog/posthog-android/blob/09940e6921bafa9e01e7d68b8c9032671a21ae73/posthog/src/main/java/com/posthog/android/PostHog.java#L278
+    // android: https://github.com/PostHog/posthog-android/blob/09940e6921bafa9e01e7d68b8c9032671a21ae73/posthog/src/main/java/com/posthog/android/PostHogActivityLifecycleCallbacks.java#L126
+
+    const prevAppBuild = this.getPersistedProperty(PostHogPersistedProperty.InstalledAppBuild)
+    const prevAppVersion = this.getPersistedProperty(PostHogPersistedProperty.InstalledAppVersion)
+    const appBuild = this._appProperties.$app_build
+    const appVersion = this._appProperties.$app_version
+
+    if (!appBuild || !appVersion) {
+      console.warn(
+        'PostHog could not track installation/update/open, as the build and version were not set. ' +
+          'This can happen if some dependencies are not installed correctly, or if you have provided' +
+          'customAppProperties but not included $app_build or $app_version.'
+      )
+    }
+    if (appBuild) {
+      if (!prevAppBuild) {
+        // new app install
+        this.capture('Application Installed', {
```

Comment:
This function is only called when people explicitly set `options?.captureNativeAppLifecycleEvents`, so anyone doing an update without reading the docs and making code changes just won't get these events.

Maybe one slight optimization would be storing the build number in storage regardless, so that people enabling this flag in future don't get a flood of events.

Reply:
Persistance now happens regardless